File: frontend/src/app/core/models/category.ts
import { TransactionType } from './transaction-type';

export interface Category {
  id: string;
  name: string;
  type: TransactionType;
  /** Name of a Material Symbol, e.g. "shopping_cart". */
  icon: string;
  /** Hex triplet, e.g. "#EF6C00". */
  color: string;
}

export interface CreateCategoryRequest {
  name: string;
  type: TransactionType;
  icon: string;
  color: string;
}

/** The type cannot be changed once a category exists (transactions and budgets depend on it). */
export interface UpdateCategoryRequest {
  name: string;
  icon: string;
  color: string;
}

File: backend/src/Pfm.Domain/Entities/Category.cs
namespace Pfm.Domain.Entities;

/// <summary>
/// A bucket that transactions are assigned to, for example "Miete" or "Gehalt".
/// A category is either for income or for expenses, never for both.
/// </summary>
public sealed class Category
{
    /// <summary>Assigned by the persistence layer when the category is first stored.</summary>
    public string Id { get; set; } = string.Empty;

    public required string Name { get; set; }

    public required TransactionType Type { get; set; }

    /// <summary>Name of a Material Symbol, for example <c>shopping_cart</c>.</summary>
    public required string Icon { get; set; }

    /// <summary>Accent colour as an uppercase hex triplet, for example <c>#EF6C00</c>.</summary>
    public required string Color { get; set; }
}
